src/main/kotlin/no/nav/syfo/consumer/arbeidsgiverNotifikasjonProdusent/ArbeidsgiverNotifikasjonProdusentConsumer.kt

```diff
@@ -0,0 +1,122 @@
+package no.nav.syfo.consumer.arbeidsgiverNotifikasjonProdusent
+
+import com.fasterxml.jackson.databind.DeserializationFeature
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
+import com.fasterxml.jackson.module.kotlin.registerKotlinModule
+import io.ktor.client.*
+import io.ktor.client.call.*
+import io.ktor.client.engine.cio.*
+import io.ktor.client.features.json.*
+import io.ktor.client.request.*
+import io.ktor.client.statement.*
+import io.ktor.http.*
+import kotlinx.coroutines.runBlocking
+import no.nav.syfo.UrlEnv
+import no.nav.syfo.auth.AzureAdTokenConsumer
+import no.nav.syfo.consumer.narmesteLeder.NarmesteLederConsumer
+import org.slf4j.LoggerFactory
+
+open class ArbeidsgiverNotifikasjonProdusentConsumer(urlEnv: UrlEnv, azureAdTokenConsumer: AzureAdTokenConsumer, narmesteLederConsumer: NarmesteLederConsumer) {
+    private val client: HttpClient
+    private val azureAdTokenConsumer: AzureAdTokenConsumer
+    private val narmesteLederConsumer: NarmesteLederConsumer
+    private val arbeidsgiverNotifikasjonProdusentBasepath: String
+    private val log = LoggerFactory.getLogger("no.nav.syfo.consumer.AgNotifikasjonProdusentConsumer")
+    private val scope = urlEnv.arbeidsgiverNotifikasjonProdusentApiScope
+    private val dineSykmeldteUrl = urlEnv.baseUrlDineSykmeldte
+    private val EMAIL_BODY = EMAIL_BODY_START + dineSykmeldteUrl + EMAIL_BODY_END
+
+    init {
+        client = HttpClient(CIO) {
+            expectSuccess = false
+            install(JsonFeature) {
+                serializer = JacksonSerializer {
+                    registerKotlinModule()
+                    registerModule(JavaTimeModule())
+                    configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+                }
+            }
+        }
+        this.azureAdTokenConsumer = azureAdTokenConsumer
+        this.narmesteLederConsumer = narmesteLederConsumer
+        arbeidsgiverNotifikasjonProdusentBasepath = urlEnv.arbeidsgiverNotifikasjonProdusentApiUrl
+    }
+
+    open fun createNewNotificationForArbeidsgiver(varselId: String, virksomhetsnummer: String, ansattFnr: String): String? {
+        val narmesteLederRelasjon = runBlocking { narmesteLederConsumer.getNarmesteLeder(ansattFnr, virksomhetsnummer)?.narmesteLederRelasjon }
+        val narmesteLederFnr = narmesteLederRelasjon?.narmesteLederFnr
+        val narmesteLederEpostadresse = narmesteLederRelasjon?.narmesteLederEpost
+
+        if (narmesteLederFnr != null && narmesteLederEpostadresse !== null) {
```

Comment:
Jeg lurer på om det kanskje hadde vært mest ryddig å gjøre sjekken på nærmeste leder et annet sted, slik at denne klassen er en ren produsent til arbeidsgivernotifikasjoner, ref [Single Responsibility Principle](https://en.wikipedia.org/wiki/Single-responsibility_principle).

Hva med å lage en service som har ansvar for å slå opp nærmeste leder og deretter kalle denne metoden?